core/templates/dev/head/components/common-layout-directives/common-elements/promo-bar.directive.ts

```diff
@@ -36,17 +36,40 @@ angular.module('oppia').directive('promoBar', [
         function() {
           var ctrl = this;
           var isPromoDismissed = function() {
-            if (!$window.hasOwnProperty('sessionStorage')) {
-              return false;
+            try {
+              return (
+                !!angular.fromJson($window.sessionStorage.promoIsDismissed));
```

Comment:
Looking at the code ... I don't understand what this has to do with logging in?

Reply:
The user gets logged out if he was previously logged in before disabling cookies. When he tries to log in again, he'll see this modal.

Reply:
But, if you read this code, it doesn't seem to have anything to do with logging in at all. Are you sure this is related to login/logout at all ... or is this an error that shows up generally if cookies are disabled?

Reply:
For example -- does the original error arise when the user has disabled cookies, is logged out, and visits other pages of the site? Or does it also happen when the user is logged in and is browsing around the site (About page, Teach page, etc.) after disabling cookies? Trying to understand whether this issue only arises immediately after a login attempt, or whether it happens more generally, in which case a different error message might be more appropriate.

Reply:
Hm, this error seems to arise when browsing around the site. I checked the landing page, About page and Teach page. I guess it happens generally then, should perhaps say something like "We recommend to allow cookies to have a smooth browsing experience"?

Reply:
I suggest encapsulating the check for "hasOwnProperty" in its own function, and have it return false either if it is not present, or an error is thrown when checking for existence. That seems closest in spirit to the existing code.

Reply:
Interesting -- OK. Create another method as described above, but use some other test (possibly not, or in addition to, hasOwnProperty) to check the availability of sessionStorage in a foolproof way.

Reply:
Got it! I'll try that out. Thanks!